app/controllers/admins_controller.rb

```diff
@@ -55,8 +57,15 @@ def set_admin
     @admin = Admin.find(params[:id])
```

Comment:
You can use set admin for all the cases:
```
def set_admin(admin_entity = nil)
  @admin = admin_entity || Admin.find(params[:id])
end
```
So in new and create method you can do:
```
def new
  set_admin(Admin.new)
....
def create
  set_admin(Admin.new(admin_params))
  ...
```